jsonschema-core.xml

```diff
@@ -800,49 +800,291 @@
+                    SHOULD NOT forbid additional keywords, and MUST NOT forbid
+                    the "$id", "$schema", or "$vocabulary" keywords in the root schema.
+                </t>
+                <t>
+                    It is RECOMMENDED that meta-schema authors reference each vocabulary's
+                    meta-schema using the <xref target="allOf">"allOf"</xref> keyword,
+                    although other mechanisms for constructing the meta-schema may be
+                    appropriate for certain use cases.
+                </t>
+                <t>
+                    Meta-schemas MAY impose additional constraints, including describing
+                    keywords not present in any vocabulary, beyond what the meta-schemas
+                    associated with the declared vocabularies describe.  This allows for
+                    restricting usage to a subset of a vocabulary, and for validating
+                    locally defined keywords not intended for re-use.
+                </t>
+                <t>
+                    Meta-schemas intended for local use, with no need to test for
+                    vocabulary support in arbitrary implementations, can safely omit
+                    "$vocabulary" entirely.
+                </t>
+            </section>
+            <section title="The JSON Schema Core Vocabulary">
+                <t>
+                    Keywords declared in in this specification that begin with "$" make up
+                    the JSON Schema Core vocabulary.  These keywords are either required in
+                    order process any schema or meta-schema, including those split across
+                    multiple documents, or exist to reserve keywords for purposes that
+                    require guaranteed interoperability.
+                </t>
+                <t>
+                    The Core vocabulary MUST be considered mandatory at all times, in order
+                    to bootstrap the processing of further vocabularies.  Meta-schemas
+                    that use "$vocabulary" MUST explicitly list the Core vocabulary,
+                    which MUST have a value of true indicating that it is required.
+                </t>
+                <t>
+                    The behavior of a false value for this vocabulary (and only this
+                    vocabulary) is undefined, as is the behavior when "$vocabulary"
+                    is present but the Core vocabulary is not included.  However, it
+                    is RECOMMENDED that implementations detect these cases and raise
+                    an error when they occur.
+                </t>
+                <t>
+                    Meta-schemas that do not use "$vocabulary" MUST be considered to
+                    require the Core vocabulary just as if "$vocabulary" were used
+                    with only the Core vocabulary present with a value of true.
```

Comment:
Good point here.

Maybe to add some backward compatibility, it may be additionally stated that if `$vocabulary` is not present implementations MAY try to recognize a supported vocabulary from the value of `$id`.

Reply:
@Relequestual it might be good to have a "MAY assume..." in there.  As written, the intent is that without `$vocabulary` everything works exactly as it does today.  In practice, most implementations assume the vocabularies from the core and validation specs (which is a larger number of vocabularies than strictly the core and validation assertion vocabulary).

@jgonzalezdr saying that they MAY try to do something based on the meta-schema's `$id` is the same as tying the behavior to the value of the schema's `$schema`, _which is exactly what already happens_.  We do not need to re-state it.  It's covered in the section on `$schema`.

Reply:
To @Relequestrial's comment:

I don't think that the validation vocabulary need be required, even if a schema is for validation.

A vocabulary is a just set of keywords. If I want to create a validation vocabulary with a completely disjoint set of keywords, I shouldn't be required to also include the standard set.  Perhaps my vocabulary *isn't* disjoint. Maybe it overlaps and redefines a number of the keywords. I'm that case, it would be explicitly wrong of me to include the standard set due to conflicts.

Reply:
@handrews: Do you mean that this is covered by the last paragraph of the 'The "$schema" Keyword' section?

Reply:
@gregsdennis the assumptions are only allowed when `$vocabulary` is **completely absent**.  Currently, there is no `$vocabulary` and pretty much ever implementation of which I'm aware assumes that schemas are using the Core and Validation specifications.  So this is just leaving that behavior in place.

If you redefine a standard keywords in your custom vocabulary, then `$vocabulary` will be present, and no assumptions will be made.

Reply:
@jgonzalezdr @gregsdennis I tried to address this in the latest commit, although I'm not sure the wording's really great.  It might be better to not nail this down, even, and see what people do (given that implementations have no trouble making these assumptions right now without guidance).

Reply:
@gregsdennis when I said "that is for validation", I meant "that want to use the validation vocabulary keywords".

Reply:
I'm happy with the change, and I agree @handrews that we shouldn't nail it down completly, and leave a little flexibility for any ways it might need nailing down later.
As I authored this comment, I'm resolving it.